authorize_helper.go

```diff
@@ -175,3 +182,118 @@ func IsLocalhost(redirectURI *url.URL) bool {
 	hn := redirectURI.Hostname()
 	return strings.HasSuffix(hn, ".localhost") || hn == "127.0.0.1" || hn == "localhost"
 }
+
+func WriteAuthorizeFormPostResponse(redirectURL string, parameters url.Values, rw io.Writer) {
+	t := template.Must(template.New("form_post").Parse(`<html>
+   <head>
+      <title>Submit This Form</title>
+   </head>
+   <body onload="javascript:document.forms[0].submit()">
+      <form method="post" action="{{ .RedirURL }}">
+         {{ range $key,$value := .Parameters }}
+		<input type="hidden" name="{{$key}}" value="{{index $value 0}}"/>
+         {{ end }}
+      </form>
+   </body>
+</html>`))
+
+	_ = t.Execute(rw, struct {
+		RedirURL   string
+		Parameters url.Values
+	}{
+		RedirURL:   redirectURL,
+		Parameters: parameters,
+	})
+}
+func ParseFormPostResponse(redirectURL string, resp io.ReadCloser) (authorizationCode, stateFromServer, iDToken string, token goauth.Token, rFC6749Error RFC6749Error, err error) {
```

Comment:
Is this only used for testing?

Reply:
Yes. It is only used in tests.

Reply:
In that case I would prefer moving this to a `_test` file!

Reply:
This helper function is used in tests in `fosite` package and `integration_test` package. The test helper functions declared in `_test` file are only visible within the same package. Any idea of how a test helper function can be shared between packages?

Reply:
Done